managed/services/uievents/uievents.go

```diff
@@ -0,0 +1,262 @@
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+// Package uievents provides facility to store UI events.
+package uievents
+
+import (
+	"context"
+	"encoding/json"
+	"sort"
+	"time"
+
+	"github.com/HdrHistogram/hdrhistogram-go"
+	pmmv1 "github.com/percona-platform/saas/gen/telemetry/events/pmm"
+	"github.com/sirupsen/logrus"
+
+	uievents "github.com/percona/pmm/api/uieventspb"
+	"github.com/percona/pmm/managed/services/telemetry"
+)
+
+const (
+	cleanupAfterHours    = 30
+	cleanupCheckInterval = 1 * time.Minute
```

Comment:
if we need to clean up every 30 hours, no reason to check it every 1 minute. we can use a ticker with 30 hours interval

Reply:
in that case, we don't need extra checks in `cleanupIfNeeded`

Reply:
From a performance standpoint, it is nothing. The idea here was to be able to modify cleanupAfterHours if needed or move it to config.